src/freenet/support/JVMVersion.java

```diff
@@ -34,7 +36,12 @@
 	    Pattern.compile("^0*(\\d+)(?:\\.0*(\\d+)(?:\\.0*(\\d+)(?:[_.]0*(\\d+))?)?)?.*$");
 
 	public static boolean isEOL() {
-		return isEOL(getCurrent());
+		return !isAndroid() // on android the version checks are done on the App level, so we do not check here.
+			&& isEOL(getCurrent());
+	}
+
+	public static boolean isAndroid() {
```

Comment:
Is there any reason this is `public`?

Reply:
I believe it's because isEOL is public static as well.

Reply:
May be it could be removed and in-lining Platform.isAndroid()

Reply:
I’d be content with the method being `private`… but you’re right, with a static import we could totally skip adding this method.

Reply:
There is no good reason this is public, no. The real reason is that extract-method made it public by default and I didn’t correct that.

The reason why I used a method is to preserve the structure of the isEOL method, since all other calls used in isEOL() are methods, too.